usermods/sht/usermod_sht.h

```diff
@@ -0,0 +1,312 @@
+      JsonArray jsonHumidity = user.createNestedArray("Humidity");
+
+      if (shtLastTimeUpdated == 0 || !shtReadDataSuccess) {
+        jsonTemp.add(0);
+        jsonHumidity.add(0);
+        if (shtLastTimeUpdated == 0) {
+          jsonTemp.add(" Not read yet");
+          jsonHumidity.add(" Not read yet");
+        }
+        else {
+          jsonTemp.add(" Error");
+          jsonHumidity.add(" Error");
+        }
+
+        return;
+      }
+
+      jsonHumidity.add(shtCurrentHumidity);
+      jsonHumidity.add(" RH");
+
+      jsonTemp.add(shtCurrentTemp);
+      jsonTemp.add(" °C");
+    }
+
+    void publishTemperatureAndHumidityViaMqtt() {
+      if (!WLED_MQTT_CONNECTED) return;
+      char buf[128];
+
+      sprintf_P(buf, PSTR("%s/temperature"), mqttDeviceTopic);
+      mqtt->publish(buf, 0, false, String(shtCurrentTemp).c_str());
+      sprintf_P(buf, PSTR("%s/humidity"), mqttDeviceTopic);
+      mqtt->publish(buf, 0, false, String(shtCurrentHumidity).c_str());
+    }
+
+    void publishHomeAssistantAutodiscovery() {
+      if (!WLED_MQTT_CONNECTED) return;
+
+      char json_str[1024], buf[128];
+      size_t payload_size;
+      StaticJsonDocument<1024> json;
+
+      sprintf_P(buf, PSTR("%s Temperature"), serverDescription);
+      json[F("name")] = buf;
+      sprintf_P(buf, PSTR("%s/temperature"), mqttDeviceTopic);
+      json[F("stat_t")] = buf;
+      json[F("dev_cla")] = F("temperature");
+      json[F("stat_cla")] = F("measurement");
+      sprintf_P(buf, PSTR("%s-temperature"), escapedMac.c_str());
+      json[F("uniq_id")] = buf;
+      json[F("unit_of_meas")] = F("°C");
+      appendDeviceToMqttDiscoveryMessage(json);
+      payload_size = serializeJson(json, json_str);
+      sprintf_P(buf, PSTR("homeassistant/sensor/%s/%s-temperature/config"), escapedMac.c_str(), escapedMac.c_str());
+      mqtt->publish(buf, 0, true, json_str, payload_size);
+
+      json.clear();
+
+      sprintf_P(buf, PSTR("%s Humidity"), serverDescription);
+      json[F("name")] = buf;
+      sprintf_P(buf, PSTR("%s/humidity"), mqttDeviceTopic);
+      json[F("stat_t")] = buf;
+      json[F("dev_cla")] = F("humidity");
+      json[F("stat_cla")] = F("measurement");
+      sprintf_P(buf, PSTR("%s-humidity"), escapedMac.c_str());
+      json[F("uniq_id")] = buf;
+      json[F("unit_of_meas")] = F("%");
+      appendDeviceToMqttDiscoveryMessage(json);
+      payload_size = serializeJson(json, json_str);
+      sprintf_P(buf, PSTR("homeassistant/sensor/%s/%s-humidity/config"), escapedMac.c_str(), escapedMac.c_str());
+      mqtt->publish(buf, 0, true, json_str, payload_size);
+    }
+
+    void appendDeviceToMqttDiscoveryMessage(JsonDocument& root) {
+      JsonObject device = root.createNestedObject("dev");
+      device["ids"] = escapedMac.c_str();
+      device["name"] = serverDescription;
+      device["sw"] = versionString;
+      device["mdl"] = ESP.getChipModel();
+      device["mf"] = "espressif";
+    }
+
```

Comment:
Please add `getTemperatureC()` and `getTemperatureF()` public data exchange functions so that temperature data may be used in other usermods (check Temperature (Dallas) usermod).
You may as well add humidity functions.

Reply:
Ah yeah, sure. I actually had that on my list and forgot, incl. Fahrenheit support! Will add!